### PR TITLE
Proposed changes

### PR DESCRIPTION
**Pull request by JermyTan**
1. Address field is permanently removed

2. Introduce "set reminder" functionality (only up to parsing and creating SetReminderCommand)
    - Allows user to set/edit the "threshold" of the reminder
    - "threshold" refers to the number of days X such that the reminder/notification is activated when there is X days or less to the expiry date
    - Should add "threshold" to the glossary to explain its definition to users/developers
    - Implementation of logic needs to wait until @xiaoyu-nus has implemented the reminder feature

3. Change command-arguments format
    - Before: add /nStrawberry /f11/12/1999 /tfruit, delete 3, find apple pear, sort name
    - After: add| /nStrawberry /f11/12/1999 /tfruit, delete|3, find|apple pear, sort|name
    - Guarantee to properly extract the command and arguments portion from user input
    - Original format of arguments is left to the respective implementers to change

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AddCommand.java [hunk 1]
```diff
@@ -21,9 +21,9 @@
             + PREFIX_NAME + "NAME "
             + PREFIX_EXPIRY_DATE + "EXPIRY DATE "
             + "[" + PREFIX_TAG + "TAG]...\n"
-            + "Example: " + COMMAND_WORD + " "
+            + "Example: " + COMMAND_WORD + "| "
             + PREFIX_NAME + "Strawberry "
-            + PREFIX_EXPIRY_DATE + "11/12/1999"
+            + PREFIX_EXPIRY_DATE + "11/12/1999 "
```
Comment (febee99) [inline src/main/java/seedu/address/logic/commands/AddCommand.java]: Hi guys, will all the prefixes e.g. PREFIX_NAME also be changed to "|" later on? @liawsy @JermyTan

Comment (JermyTan) [reply src/main/java/seedu/address/logic/commands/AddCommand.java]: Yes, according to the formatting on https://ay1920s1-cs2103t-f11-2.github.io/main/UserGuide.html

File: src/test/java/seedu/address/testutil/ItemUtil.java [hunk 2]
```diff
@@ -20,14 +20,14 @@
      * Returns an add command string for adding the {@code item}.
      */
     public static String getAddCommand(Item item) {
-        return AddCommand.COMMAND_WORD + " " + getItemDetails(item);
+        return AddCommand.COMMAND_WORD + "|" + getItemDetails(item);
     }
 
     /**
      * Returns the part of command string for the given {@code person}'s details.
```
Comment (liawsy) [inline src/test/java/seedu/address/testutil/ItemUtil.java]: Need to change person to item 😝

Comment (JermyTan) [reply src/test/java/seedu/address/testutil/ItemUtil.java]: I leave it up to the implementer to update the names of the methods and its corresponding javadocs.

**Pull request merged**